core/src/ledger_cleanup_service.rs

```diff
@@ -169,6 +189,38 @@ impl LedgerCleanupService {
         Ok(new_root_receiver.try_iter().last().unwrap_or(root))
     }
 
+    /// May perform cleanup expired data in ledger store based on the specified
+    /// parameters.  If `root` - `last_purge_slot` > `purge_interval`, ledger
+    /// cleanup will further check whether there're any data to cleanup based
+    /// on the input `max_ledger_shreds`.  If it does, then cleanup will be
+    /// perfomed, and both `last_purge_slot` and `last_compact_slot` will be
+    /// updated accordingly.
+    ///
+    /// If `root` - `last_purge_slot` <= `purge_interval`, the function will
+    /// simply become no-op and return Ok status.
```

Comment:
Instead of repeating the same line as the code, `root - *last_purge_slot > purge_interval`, I feel like this comment should be higher level. My quick stab at it:
"Checks for new roots and initiates a cleanup if the last cleanup was at least `purge_interval` slots ago. A cleanup will no-op if the ledger already has fewer than `max_ledger_shreds`; otherwise, the cleanup will choose enough slots to clean to get the ledger size below `max_ledger_shreds`."